packages/edge-config/src/utils/create-loaders.ts

```diff
@@ -0,0 +1,338 @@
+    async (keys: readonly string[]) => {
+      const localEdgeConfig = await getFileSystemEdgeConfig(connection);
+
+      if (localEdgeConfig) {
+        return keys.map((key) => hasOwnProperty(localEdgeConfig.items, key));
+      }
+
+      const allItemsPromise = getAllMap.get('#');
+      if (allItemsPromise) {
+        try {
+          const allItems = await allItemsPromise;
+          return keys.map((key) => hasOwnProperty(allItems, key));
+        } catch {
+          /* ignored */
+        }
+      }
+
+      return Promise.all(
+        // TODO introduce an endpoint for batch evaluating has()
+        keys.map((key) => {
+          // this is a HEAD request anyhow, no need for fetchWithCachedResponse
+          return fetch(`${baseUrl}/item/${key}?version=${version}`, {
+            method: 'HEAD',
+            headers: new Headers(headers),
+            cache: 'no-store',
+          }).then(
+            (res) => {
+              if (res.status === 401) throw new Error(ERRORS.UNAUTHORIZED);
+              if (res.status === 404) {
+                // if the x-edge-config-digest header is present, it means
+                // the edge config exists, but the item does not
+                if (res.headers.has('x-edge-config-digest')) return false;
+                // if the x-edge-config-digest header is not present, it means
+                // the edge config itself does not exist
+                throw new Error(ERRORS.EDGE_CONFIG_NOT_FOUND);
+              }
+              if (res.ok) return true;
+              throw new Error(ERRORS.UNEXPECTED);
+            },
+            (error) => {
+              if (isDynamicServerError(error)) throw error;
+              throw new Error(ERRORS.NETWORK);
+            },
+          );
+        }),
+      );
+    },
+    {
+      batchScheduleFn,
+      // disable batching until we have an endpoint to btch evaluate "has" calls,
+      // otherwise the slowest has call will currently delay all others
+      maxBatchSize: 1,
+      cacheMap: hasMap,
+    },
+  );
+
+  const getAllLoader = new DataLoader(
+    async (keys: readonly string[]) => {
+      // as every edge config has a single "all" only, we use # as the key
+      // to load all items
+      if (keys.length !== 1 || keys[0] !== '#') {
+        throw new Error('unexpected key passed to digest');
+      }
+
+      const localEdgeConfig = await getFileSystemEdgeConfig(connection);
+
+      if (localEdgeConfig) {
+        // returns an array as "#" is the only key
+        return [localEdgeConfig.items];
+      }
+
+      const edgeConfigItems = (await fetchWithCachedResponse(
+        `${baseUrl}/items?version=${version}`,
+        {
+          headers: new Headers(headers),
+          cache: 'no-store',
+        },
+      ).then(
+        async (res) => {
+          if (res.ok) {
+            return (await res.json()) as EdgeConfigItems;
+          }
+          await consumeResponseBodyInNodeJsRuntimeToPreventMemoryLeak(res);
```

Comment:
Do we need to await these? Or could we just `void` them?

Reply:
good shout, fixed in f26318ea71eae611b0a2edf6b36c0aa0163816ac